generate.py

```diff
@@ -0,0 +1,599 @@
+    TAGS_MULTIPLIER = 1
+
+    _create_tags_recursively(
+        1, MAX_LEVELS, TAGS_MULTIPLIER,
+        two_level_taxonomy, "two level tag", parent=None
+    )
+
+
+def create_tags_for_multi_org_taxonomy(multi_org_taxonomy):
+    """
+    Create 5 tags for the multi_org_taxonomy
+    """
+    for i in range(5):
+        Tag.objects.create(
+            taxonomy=multi_org_taxonomy, value=f"multi org taxonomy tag {i}"
+        )
+
+
+def create_tags_from_json(open_canada_taxonomy, import_json_path):
+    """
+    Create tags based what is defined in JSON import spec
+    """
+
+    def _create_tags(taxonomy_data, parent):
+        if len(taxonomy_data) == 0:
+            return
+
+        for data in taxonomy_data:
+            tag = Tag.objects.create(
+                taxonomy=open_canada_taxonomy,
+                value=data.get("name"),
+                parent=parent,
+                external_id=data.get("external_id")
+            )
+            _create_tags(data.get("children"), tag)
+
+    with open(import_json_path, 'r') as json_file:
+        taxonomy_data = json.load(json_file)
+
+    _create_tags(taxonomy_data, None)
+
+
+def tagify_object(object_id, taxonomies):
+    """
+    Tag object with tags from the provided taxonomies
+
+    Arguments:
+        object_id: ID of object to be tagged
+        taxonomies: list of taxonomies of tags to tag object with
+    """
+    for taxonomy in taxonomies:
+        tag = taxonomy.get_tags()[0]
+        try:
+            tag_content_object(object_id, taxonomy, [tag.value])
+        except IntegrityError:
+            # content tag value already exists, we need to resync with
+            # new tag instance
+            content_tags = list(get_content_tags(object_id, taxonomy.id))
+            resync_object_tags(content_tags)
+
+
+# Generate sample organizations or retrieve them if they already exist
+logger.info("Generating or retrieving sample Organizations...")
+sample_orgs = []
+for i in range(1, SAMPLE_ORGS_COUNT+1):
+    org, created = Organization.objects.get_or_create(
+        name=f"{SAMPLE_ORG_NAME}{i}",
+        short_name=f"{SAMPLE_ORG_NAME}{i}"
+    )
+    logger.info(f"{'Created' if created else 'Retrieved'} {org}")
+    sample_orgs.append(org)
+
+store = modulestore()
+
+# Retrieve/Create multi org Taxonomy with 5 tags for the sample orgs
+logger.info(f"Creating or retrieving {MULTI_ORG_TAXONOMY_NAME}")
+multi_org_taxonomy = get_or_create_taxonomy(
+    None, MULTI_ORG_TAXONOMY_NAME, sample_orgs, enabled=True
+)
+
+# Clear any existing Tags for hierarchical_taxonomy and create fresh ones
+multi_org_taxonomy_tags = multi_org_taxonomy.get_tags()
+logger.info(f"Clearing existing {len(multi_org_taxonomy_tags)} Tags for {multi_org_taxonomy}")
+for tag in tqdm.tqdm(multi_org_taxonomy_tags):
+    tag.delete()
+
+logger.info(f"Creating fresh Tags for {multi_org_taxonomy}")
+multi_org_taxonomy_tags = create_tags_for_multi_org_taxonomy(multi_org_taxonomy)
+
+
+for org in sample_orgs:
```

Comment:
Could we make the LightCast taxonomy and Canada Skill taxonomies into "all orgs" taxonomies instead? Any taxonomy that's coming from an external source doesn't need a separate copy for each org.

Reply:
Got it, done.